### PR TITLE
Martin/video with alpha

### PR DESCRIPTION
**Pull request by MartinJibber**
https://user-images.githubusercontent.com/88675954/176090655-60bb0940-9457-4be7-9706-28881be11c59.mov

### CODE REVIEW

File: Shared/UI/Face Detection/FaceCaptureViewController.swift [hunk 1]
```diff
@@ -283,7 +283,7 @@ extension FaceCaptureViewController: AVCaptureVideoDataOutputSampleBufferDelegat
 
             // Create an asset writer that will write the video to the url
             self.videoWriter = try AVAssetWriter(outputURL: url, fileType: .mov)
-            let settings: [String : Any] = [AVVideoCodecKey : AVVideoCodecType.hevc,
+            let settings: [String : Any] = [AVVideoCodecKey : AVVideoCodecType.hevcWithAlpha,
```
Comment (MartinJibber) [inline Shared/UI/Face Detection/FaceCaptureViewController.swift:285]: Basically I just had to turn alpha on. Easy.

File: Shared/UI/Face Detection/FaceCaptureViewController.swift [hunk 2]
```diff
@@ -294,9 +294,15 @@ extension FaceCaptureViewController: AVCaptureVideoDataOutputSampleBufferDelegat
...
+            let pixelBufferAttributes = [
+                    kCVPixelBufferPixelFormatTypeKey: kCVPixelFormatType_32BGRA,
+                    kCVPixelBufferWidthKey: 480,
+                    kCVPixelBufferHeightKey: 480,
+                    kCVPixelBufferMetalCompatibilityKey: true] as [String: Any]
+
             self.pixelBufferAdaptor
             = AVAssetWriterInputPixelBufferAdaptor(assetWriterInput: self.videoWriterInput,
-                                                   sourcePixelBufferAttributes: nil)
+                                                   sourcePixelBufferAttributes: pixelBufferAttributes)
```
Comment (MartinJibber) [inline Shared/UI/Face Detection/FaceCaptureViewController.swift]: Improving video capture performance.

File: Shared/UI/Views/VideoView.swift [hunk 3]
```diff
@@ -34,28 +32,18 @@ class VideoView: BaseView {
...
         guard let videoURL = url else {
             self.playerLayer.player = nil
             return
         }
 
-        let player = AVPlayer(url: videoURL)
+        let videoItem = AVPlayerItem(url: videoURL)
+        let player = AVQueuePlayer(items: [videoItem])
+        self.looper = AVPlayerLooper(player: player, templateItem: videoItem)
+
```
Comment (MartinJibber) [inline Shared/UI/Views/VideoView.swift:45]: This AVPlayerLooper is a better way to loop the expression.

File: Shared/UI/Face Detection/FaceCaptureViewController.swift [hunk 4]
```diff
@@ -283,20 +282,28 @@ extension FaceCaptureViewController: AVCaptureVideoDataOutputSampleBufferDelegat
 
             // Create an asset writer that will write the video to the url
             self.videoWriter = try AVAssetWriter(outputURL: url, fileType: .mov)
-            let settings: [String : Any] = [AVVideoCodecKey : AVVideoCodecType.hevc,
+            let settings: [String : Any] = [AVVideoCodecKey : AVVideoCodecType.hevcWithAlpha,
                                             AVVideoWidthKey : 480,
                                            AVVideoHeightKey : 480,
-                            AVVideoCompressionPropertiesKey : [AVVideoQualityKey : 0.5]]
+                            AVVideoCompressionPropertiesKey : [AVVideoQualityKey : 0.5,
+                                 kVTCompressionPropertyKey_TargetQualityForAlpha : 0.5]
```
Comment (MartinJibber) [inline Shared/UI/Face Detection/FaceCaptureViewController.swift:289]: Compressing the alpha layer of the video to keep file size down.

File: Shared/UI/Face Detection/FaceCaptureViewController.swift (hunk 1 above)
Comment (Benjamindodgson) [reply Shared/UI/Face Detection/FaceCaptureViewController.swift:285]: sick

**Pull request merged**